scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/StartDeployment.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.schedulingmanager.deployment.steps;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.blox.schedulingmanager.deployment.handler.Encoder;
+import com.amazonaws.blox.schedulingmanager.deployment.steps.types.StateData;
+import com.amazonaws.blox.schedulingmanager.deployment.steps.types.TaskWorkflowInput;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.stepfunctions.AWSStepFunctions;
+import com.amazonaws.services.stepfunctions.model.StartExecutionRequest;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@AllArgsConstructor
+public class StartDeployment implements StepHandler {
+
+  public static final String TASK_WF_ARN =
+      "arn:aws:states:us-east-1:159403520677:stateMachine:StartTaskWorkflow-CPYPRFKCFCYC";
+
+  private Encoder encoder;
+  private AWSStepFunctions stepFunctions;
+
+  @Override
+  public void handleRequest(InputStream input, OutputStream output, Context context)
+      throws IOException {
+    log.debug("startDeployment lambda");
+
+    try {
+      final StateData stateData = encoder.decode(input, StateData.class);
+    } catch (final IOException e) {
+      log.error("Could not parse input into the expected class {}", StateData.class.getName());
+      throw e;
+    }
+
+    final TaskWorkflowInput taskWorkflowInput =
+        TaskWorkflowInput.builder().taskDefinition("taskDef").build();
+
+    final String taskWorkflowInputJson = encoder.encode(taskWorkflowInput);
+
+    //TODO: spawn workflows for each task
+    //TODO: don't hardcode the statemachine arn or replace accountid/region
+    final StartExecutionRequest startExecutionRequest =
+        new StartExecutionRequest()
+            .withStateMachineArn(TASK_WF_ARN)
+            .withInput(taskWorkflowInputJson)
+            .withName("StartTaskWorkflow" + UUID.randomUUID().toString());
```

Comment:
Don't we have to pass this name on into the task that waits for the execution of this workflow to complete?

Reply:
which task? ECS task?

Reply:
Ah, damn, terminology overlap. No, the Stepfunctions Task that waits for the start task workflow(s) to complete. Presumably we want to do something once the task is actually up and running? Or will we just get that notification by looking at the actual state of the cluster?

Reply:
No so this workflow just kicks off workflows and completes. The workflows themselves have waits until the task is up. If the workflow fails, the task will be started by the next reconciliation/deployment loop.